openweathermap-ios/openweathermap-ios/Maps/ViewControllers/MapViewController.swift

```diff
@@ -20,10 +24,11 @@ final class MapViewController: UIViewController {
         locationManager.desiredAccuracy = kCLLocationAccuracyBest
         locationManager.requestWhenInUseAuthorization()
         locationManager.startUpdatingLocation()
+        weatherManager.delegate = self
     }
     
-    func updateAnnotation(location: CLLocationCoordinate2D) {
-        let annotation = Annotation(coordinate: location, title: "Rio Grande", subtitle: "Mostly Cloudy")
+    func updateAnnotation(location: CLLocationCoordinate2D, city: String, description: String) {
```

Comment:
Why are you adding parameters here if you have access directly by the self?

Reply:
I don't know how to do it any other way.

Reply:
Can you help me by giving a direction?

Reply:
Just remove the parameters and use the properties directly.

```
let annotation = Annotation(
    coordinate: location, 
    title: cityName, 
    subtitle: descriptionWeather
)
```